### PR TITLE
[Cupertino] fix dark mode for `ContextMenuAction`

### LINKED ISSUE #43211: Cupertino ContextMenu Widget Dark Mode Support
Cupertino's ContextMenu widget, introduced in https://github.com/flutter/flutter/pull/37778, does not support dark mode.

It has two obvious colors that change with dark mode / light mode that I'm aware of.  When the child is pressed, it has a brief flash of light or dark color, which is currently hardcoded as just `_lightModeMaskColor`.  Also, when the ContextMenu is open, the ContextMenuAction items should adjust their color based on the mode.

| light | dark |
| ---- | ----- |
| context_menu_mask_light | context_menu_mask_dark |
| Screen Shot 2019-10-21 at 1 46 35 PM | Screen Shot 2019-10-21 at 1 48 29 PM |

### PR DESCRIPTION
**Pull request by TahaTesser**
It partially fixes https://github.com/flutter/flutter/issues/43211 (dark theme now works tho flash effect is pending)

### Preview

https://user-images.githubusercontent.com/48603081/138452610-8cbfdc1b-4ae4-4ca0-a358-98b1b27dfbb1.mp4

| iOS | Flutter  |
| --------------- | --------------- |
|  |  |

| iOS | Flutter  |
| --------------- | --------------- |
|  |  |

Of course, `CupertinoContextMenu` appearance isn't exactly like native iOS. but that's unrelated to this PR. See https://github.com/flutter/flutter/issues/92260 for more details.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/cupertino/context_menu_action.dart [hunk 1]
```diff
@@ -114,7 +122,9 @@ class _CupertinoContextMenuActionState extends State<CupertinoContextMenuAction>
           button: true,
           child: Container(
             decoration: BoxDecoration(
-              color: _isPressed ? _kBackgroundColorPressed : _kBackgroundColor,
+              color: _isPressed
+              ? CupertinoDynamicColor.resolve(_kBackgroundColorPressed, context)
```
Comment (HansMuller) [inline packages/flutter/lib/src/cupertino/context_menu_action.dart:126]: two space indent

File: packages/flutter/lib/src/cupertino/context_menu_action.dart [hunk 2]
```diff
@@ -93,7 +99,9 @@ class _CupertinoContextMenuActionState extends State<CupertinoContextMenuAction>
         color: CupertinoColors.destructiveRed,
       );
     }
-    return _kActionSheetActionStyle;
+    return _kActionSheetActionStyle.copyWith(
+        color: CupertinoDynamicColor.resolve(CupertinoColors.label, context)
```
Comment (HansMuller) [inline packages/flutter/lib/src/cupertino/context_menu_action.dart:103]: two space indent

File: packages/flutter/test/cupertino/context_menu_action_test.dart [hunk 3]
```diff
@@ -5,15 +5,24 @@
...
+  // static const Color _kBackgroundColorPressed = Color(0xFFDDDDDD);
+  const CupertinoDynamicColor _kBackgroundColorPressed = CupertinoDynamicColor.withBrightness(
+    color: Color(0xFFDDDDDD),
+    darkColor: Color(0xFF3F3F40),
+  );
   const Color _kDestructiveActionColor = CupertinoColors.destructiveRed;
   const FontWeight _kDefaultActionWeight = FontWeight.w600;
 
-  Widget _getApp({VoidCallback? onPressed, bool isDestructiveAction = false, bool isDefaultAction = false}) {
+  Widget _getApp({VoidCallback? onPressed, bool isDestructiveAction = false,
```
Comment (HansMuller) [inline packages/flutter/test/cupertino/context_menu_action_test.dart:24]: Typical formatting would be:
```dart
Widget _getApp({
  VoidCallback? onPressed, 
  bool isDestructiveAction = false,
  bool isDefaultAction = false, 
  bool isDark = false,
}) {
```

**Review by HansMuller — COMMENTED** [review]
Looks good, just some trivial feedback about formatting

**Comment by TahaTesser** [comment]
@HansMuller 
Thanks so much for the feedback, appreciate it, 
This looks good to go, Let me know if anything is needed.

**Comment by TahaTesser** [comment]
@HansMuller 
Does the conflict need to be resolved here?

**Comment by jmagman** [comment]
> @HansMuller Does the conflict need to be resolved here?

@TahaTesser yes, you need to resolve the conflicts by rebasing on or merging in master.  You can follow the "Use the command line" instructions below the warning.

**Comment by TahaTesser** [comment]
@jmagman @HansMuller 
Fixed conflict, thanks!

**Comment by TahaTesser** [comment]
@HansMuller 
Looks like it won't merge one more approve

File: packages/flutter/test/cupertino/context_menu_action_test.dart [hunk 4]
```diff
@@ -25,7 +25,7 @@ void main() {
     VoidCallback? onPressed,
     bool isDestructiveAction = false,
     bool isDefaultAction = false,
-    bool isDark = false,
+    Brightness? brightness,
```
Comment (TahaTesser) [inline packages/flutter/test/cupertino/context_menu_action_test.dart:28]: @HansMuller 
Made this slight change, this looks cleaner than isDark variable

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by servonlewis** [comment]
Any idea when this will be merged and added to Flutter?

**Comment by TahaTesser** [comment]
@servonlewis 
The fix is currently available in Flutter 2.9+, It should land soon on the stable channel soon, thank you!

**Comment by servonlewis** [comment]
> @servonlewis 
> 
> The fix is currently available in Flutter 2.9+, It should land on the stable channel soon, thank you!

Thank you!

**Pull request merged**